great_expectations/rule_based_profiler/config/base.py

```diff
@@ -237,51 +230,52 @@ def __init__(
         self,
         name: str,
         config_version: float,
-        rules: Dict[str, RuleConfig],
+        rules: Dict[str, dict],  # see RuleConfig
         class_name: str,
         module_name: Optional[str] = None,
         variables: Optional[Dict[str, Any]] = None,
         commented_map: Optional[CommentedMap] = None,
-        **kwargs
     ):
         self.name = name
         self.config_version = config_version
         self.rules = rules
         self.class_name = class_name
         self.module_name = module_name
         self.variables = variables
-        for k, v in kwargs.items():
-            setattr(self, k, v)
-            logger.warning(
-                "Setting unknown kwarg (%s, %s) provided to config as attr", k, v
-            )
 
         super().__init__(commented_map=commented_map)
 
     @classmethod
-    def get_config_class(cls) -> Type["RuleBasedProfilerConfig"]:
+    def get_config_class(cls) -> Type["RuleBasedProfilerConfig"]:  # noqa: F821
         return cls
 
     @classmethod
-    def get_schema_class(cls) -> Type["RuleBasedProfilerConfigSchema"]:
+    def get_schema_class(cls) -> Type["RuleBasedProfilerConfigSchema"]:  # noqa: F821
         return RuleBasedProfilerConfigSchema
 
 
-class RuleBasedProfilerConfigSchema(NotNullSchema):
+class RuleBasedProfilerConfigSchema(Schema):
```

Comment:
If this now uses `Schema`, what is the purpose of `__config_class__`? Let's define a method that does the `@post_load` behavior that `NotNullSchema` implicitly did for us?

If we're keeping in the class variable, please add a comment reflecting it's relevance to this overall config file.

Reply:
But do we want a method to point to our config? 

```python
def make_rule_based_profiler_config(self, data, **kwargs):
    return RuleBasedProfilerConfig(**data)
```